src/services/companyProfileService.ts

```diff
@@ -1,31 +1,33 @@
-import { Resource, createApiClient } from "@companieshouse/api-sdk-node";
+import { Resource } from "@companieshouse/api-sdk-node";
+import ApiClient from "@companieshouse/api-sdk-node/dist/client";
 import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
-import { Session } from "@companieshouse/node-session-handler";
+import { HttpStatusCode } from "axios";
 import { Request } from "express";
-import { logger } from "../lib/logger";
-import { getAccessToken } from "../utils/session";
+import { createOAuthApiClient } from "./apiClientService";
+import { createAndLogError, logger } from "../lib/logger";
+import { ApiErrorResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
 
-export const getCompanyProfile = async (req: Request, companyNumber: string): Promise<CompanyProfile> => {
+export const getCompanyProfile = async (request: Request, companyNumber: string): Promise<CompanyProfile> => {
+    const oAuthApiClient: ApiClient = createOAuthApiClient(request.session);
 
-    const accessToken: string = getAccessToken(req.session as Session);
-    const apiClient = createApiClient(undefined, accessToken, undefined);
-
-    logger.debug(`${getCompanyProfile.name} - Looking for company profile with company number ${companyNumber}`);
-    const sdkResponse: Resource<CompanyProfile> = await apiClient.companyProfile.getCompanyProfile(companyNumber);
+    logger.debug(`${getCompanyProfile.name} - Get company profile with company number ${companyNumber}`);
+    const sdkResponse: Resource<CompanyProfile> | ApiErrorResponse = await oAuthApiClient.companyProfile.getCompanyProfile(companyNumber);
 
     if (!sdkResponse) {
-        throw logger.info(`${getCompanyProfile.name} - Company Profile API returned no response for company number ${companyNumber}`);
+        throw createAndLogError(`${getCompanyProfile.name} - Company Profile API returned no response for company number ${companyNumber}`);
     }
 
```

Comment:
We could provide more detail about the errors, if you think that's useful, something like:

```suggestion
    if (sdkResponse.httpStatusCode === HttpStatusCode.InternalServerError && "errors" in sdkResponse) {
        const messages = sdkResponse.errors?.map(e => e.error);
        throw createAndLogError(`${getCompanyProfile.name} - Company Profile API returned with errors for company number ${companyNumber}: ${messages}`);
    }
```